site/content/en/docs/Tasks/configmaps_and_secrets.md

```diff
@@ -1,8 +1,356 @@
+
+1. Create a Kustomization file.
+```yaml
+# kustomization.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+configMapGenerator:
+- name: my-java-server-env-vars
+  literals:
+  - JAVA_HOME=/opt/java/jdk
+  - JAVA_TOOL_OPTIONS=-agentlib:hprof
+resources:
+- deployment.yaml
+```
+
+2. Create a Deployment manifest.
+```yaml
+# deployment.yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: test-deployment
+  labels:
+    app: test
+spec:
+  selector:
+    matchLabels:
+      app: test
+  template:
+    metadata:
+      labels:
+        app: test
+    spec:
+      containers:
+      - name: container
+        image: registry.k8s.io/busybox
+        command: [ "/bin/sh", "-c", "ls /etc/config/" ]
+        volumeMounts:
+        - name: config-volume
+          mountPath: /etc/config
+      volumes:
+      - name: config-volume
+        configMap:
+          name: my-java-server-env-vars
+```
+
+3. Create the ConfigMap using `kustomize build`.
+```bash
+kustomize build .
+```
+
+The output is similar to:
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  # The name has been updated to include the suffix
+  name: my-java-server-env-vars-k44mhd6h5f
+data:
+  JAVA_HOME: /opt/java/jdk
+  JAVA_TOOL_OPTIONS: -agentlib:hprof
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: test
+  name: test-deployment
+spec:
+  selector:
+    matchLabels:
+      app: test
+  template:
+    metadata:
+      labels:
+        app: test
+    spec:
+      containers:
+      - command:
+        - /bin/sh
+        - -c
+        - ls /etc/config/
+        image: registry.k8s.io/busybox
+        name: container
+        volumeMounts:
+        - mountPath: /etc/config
+          name: config-volume
+      volumes:
+      - configMap:
+          # The name has been updated to include the
+          # suffix matching the ConfigMap
+          name: my-java-server-env-vars-k44mhd6h5f
+        name: config-volume
+```
+
+
+## Create a TLS Secret
+
+The following example generates a TLS Secret with certificate and private key data files.
+
+1. Create a Kustomization file.
+```yaml
+# kustomization.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+secretGenerator:
+- name: app-tls
+  files:
+    - "tls.crt"
+    - "tls.key"
+  type: "kubernetes.io/tls"
+```
+
+2. Create a certificate file.
+```yaml
+# tls.crt
+LS0tLS1CRUd...tCg==
+```
+
+3. Create a private key file.
+```yaml
+# tls.key
+LS0tLS1CRUd...0tLQo=
+```
+
+4. Create the Secret using `kustomize build`.
+```bash
+kustomize build .
+```
+
+The output is similar to:
+```yaml
+apiVersion: v1
+data:
+  tls.crt: TFMwdExTMUNSVWQuLi50Q2c9PQ==
+  tls.key: TFMwdExTMUNSVWQuLi4wdExRbz0=
+kind: Secret
+metadata:
+  name: app-tls-c888dfbhf8
+type: kubernetes.io/tls
+```
+
+{{< alert color="warning" title="Important" >}}
+It is important to note that the secrets are `base64` encoded.
+{{< /alert >}}
+
+
+## Roll out updates
```

Comment:
I don't think we need this section. We share the existing website https://kubectl.docs.kubernetes.io/guides/config_management/secrets_configmaps/ with `kubectl`, which is why it contains information about `kubectl apply` and how `kubectl` handles changes to `ConfigMap`s.

If you'd like, we can use part of this explanation under `Concepts` to explain why `configMapGenerator` appends a hash suffix to generated `ConfigMap` names.

Reply:
No problem. I removed this section.